From 473a9e8b0022595809a1af673091a0b387b0cb86 Mon Sep 17 00:00:00 2001
From: Yan Fu <yafu@redhat.com>
Date: Tue, 21 Oct 2025 23:04:54 -0400
Subject: [PATCH 6/7] Fedora: remove jwt-simple dependency
Content-type: text/plain

jwt-simple is not in Fedora.

The code checks time validity. Use std::time instead for this.

Signed-off-by: Uri Lublin <uril@redhat.com>
Signed-off-by: Yan Fu <yafu@redhat.com>
---
 Cargo.toml                                     |  3 ---
 attestation-agent/kbs_protocol/Cargo.toml      |  1 -
 .../kbs_protocol/src/token_provider/mod.rs     | 18 ++++++++----------
 3 files changed, 8 insertions(+), 14 deletions(-)

diff --git a/Cargo.toml b/Cargo.toml
index ebf97fd..8b1d61e 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -32,9 +32,6 @@ config = "0.15.23"
 const_format = "0.2.36"
 hex = "0.4.3"
 hmac = "0.13.0"
-jwt-simple = { version = "0.12", default-features = false, features = [
-    "pure-rust",
-] }
 kbs-types = "0.18.0"
 nix = "0.31"
 openssl = "0.10"
diff --git a/attestation-agent/kbs_protocol/Cargo.toml b/attestation-agent/kbs_protocol/Cargo.toml
index 6e38a09..9393ada 100644
--- a/attestation-agent/kbs_protocol/Cargo.toml
+++ b/attestation-agent/kbs_protocol/Cargo.toml
@@ -15,7 +15,6 @@ base64.workspace = true
 serde_json_canonicalizer = { version = "0.3", optional = true }
 clap = { workspace = true, features = ["derive"], optional = true }
 crypto = { path = "../deps/crypto", default-features = false }
-jwt-simple.workspace = true
 kbs-types.workspace = true
 reqwest = { workspace = true, default-features = false, features = [
     "cookies",
diff --git a/attestation-agent/kbs_protocol/src/token_provider/mod.rs b/attestation-agent/kbs_protocol/src/token_provider/mod.rs
index 741e83c..001decb 100644
--- a/attestation-agent/kbs_protocol/src/token_provider/mod.rs
+++ b/attestation-agent/kbs_protocol/src/token_provider/mod.rs
@@ -14,11 +14,8 @@ pub use aa::*;
 use anyhow::*;
 use async_trait::async_trait;
 use base64::{Engine, engine::general_purpose::URL_SAFE_NO_PAD};
-use jwt_simple::{
-    claims::JWTClaims,
-    prelude::{Clock, UnixTimeStamp},
-};
 use serde_json::Value;
+use std::time::SystemTime;
 
 use crate::keypair::TeeKeyPair;
 
@@ -33,8 +30,8 @@ pub trait TokenProvider: Send + Sync {
 #[derive(Clone, Debug)]
 pub struct Token {
     pub content: String,
-    exp: Option<UnixTimeStamp>,
-    nbf: Option<UnixTimeStamp>,
+    exp: Option<u64>,
+    nbf: Option<u64>,
 }
 
 impl Token {
@@ -44,16 +41,17 @@ impl Token {
             .nth(1)
             .ok_or_else(|| anyhow!("illegal token format"))?;
         let claims = URL_SAFE_NO_PAD.decode(claims_b64)?;
-        let claims = serde_json::from_slice::<JWTClaims<Value>>(&claims)?;
+        let claims = serde_json::from_slice::<Value>(&claims)?;
         Ok(Self {
             content: token,
-            exp: claims.expires_at,
-            nbf: claims.invalid_before,
+            exp: claims["exp"].as_u64(),
+            nbf: claims["nbf"].as_u64(),
         })
     }
 
     pub fn check_valid(&self) -> Result<()> {
-        let now = Clock::now_since_epoch();
+        let now = SystemTime::now().duration_since(SystemTime::UNIX_EPOCH)
+            .unwrap().as_secs();
         if let Some(exp) = self.exp
             && exp < now
         {
-- 
2.0
